randovania/generator/base_patches_factory.py

```diff
@@ -6,259 +6,123 @@
 from randovania.game_description.game_patches import GamePatches
-from randovania.game_description.hint import Hint, HintType, PrecisionPair, HintLocationPrecision, HintItemPrecision, \
-    HintDarkTemple
-from randovania.game_description.requirements import RequirementAnd, ResourceRequirement, Requirement
-from randovania.game_description.resources import search
-from randovania.game_description.resources.pickup_index import PickupIndex
-from randovania.game_description.resources.resource_type import ResourceType
 from randovania.game_description.world.area_identifier import AreaIdentifier
-from randovania.game_description.world.node import LogbookNode, LoreType, ConfigurableNode
 from randovania.game_description.world.world_list import WorldList
-from randovania.games.game import RandovaniaGame
-from randovania.games.prime2.layout.echoes_configuration import EchoesConfiguration
-from randovania.games.prime2.layout.translator_configuration import LayoutTranslatorRequirement
 from randovania.generator import elevator_distributor
+from randovania.layout.base.base_configuration import BaseConfiguration
 from randovania.layout.lib.teleporters import TeleporterShuffleMode
-from randovania.lib.enum_lib import iterate_enum
 
 
 class MissingRng(Exception):
     pass
 
-
-def add_elevator_connections_to_patches(configuration: EchoesConfiguration,
-                                        rng: Random,
-                                        patches: GamePatches) -> GamePatches:
-    """
-    :param configuration:
-    :param rng:
-    :param patches:
-    :return:
-    """
-    elevator_connection = copy.copy(patches.elevator_connection)
-    elevators = configuration.elevators
-
-    if not elevators.is_vanilla:
-        if rng is None:
-            raise MissingRng("Elevator")
-
-        world_list = default_database.game_description_for(configuration.game).world_list
-        elevator_db = elevator_distributor.create_elevator_database(
-            world_list, elevators.editable_teleporters)
-
-        if elevators.mode in {TeleporterShuffleMode.TWO_WAY_RANDOMIZED, TeleporterShuffleMode.TWO_WAY_UNCHECKED}:
-            connections = elevator_distributor.two_way_elevator_connections(
-                rng=rng,
-                elevator_database=elevator_db,
-                between_areas=elevators.mode == TeleporterShuffleMode.TWO_WAY_RANDOMIZED
-            )
+class BasePatchesFactory:
+    def create_base_patches(self,
+                            configuration: BaseConfiguration,
+                            rng: Random,
+                            game: GameDescription,
+                            is_multiworld: bool,
+                            player_index: int,
+                            ) -> GamePatches:
+        """
+        """
+        patches = dataclasses.replace(game.create_game_patches(),
+                                    player_index=player_index)
+
+        # Elevators
+        if hasattr(configuration, "elevators"):
```

Comment:
Th `self.add_elevator_connections_to_patches` should just always be called.

Reply:
:+1: